ncm-spma/src/test/perl/apt-cleanup-sources.t

```diff
@@ -0,0 +1,117 @@
+# -*- mode: cperl -*-
+# ${license-info}
+# ${author-info}
+# ${build-info}
+
+=pod
+
+=head1 DESCRIPTION
+
+Tests for the cleanup_old_sources method.  This method ensures that
+outdated sources in /etc/apt/sources.list.d/ are removed.
+
+=head1 TESTS
+
+=cut
+
+use strict;
+use warnings;
+use Readonly;
+use English;
+
+use Test::Quattor;
+use Test::More;
+use Test::MockModule;
+use NCM::Component::spma::apt;
+use CAF::Path;
+use CAF::FileWriter;
+use Set::Scalar;
+
+
+Readonly my $SOURCE_DIR => "target/test/clean.sources.list.d";
+
+my $sources = [{
+    name => "source1",
+    owner => 'localuser@localdomain',
+    protocols => [{
+        name => "http",
+        url => "http://localhost.localdomain",
+    }],
+}];
+
+my $cmp = NCM::Component::spma::apt->new("spma");
+my $mock = Test::MockModule->new('NCM::Component::spma::apt');
+
+my $immutable;
+my $directory_exists;
+my $sourcefiles;
+
+
+$mock->mock('_glob', sub {
+    return @$sourcefiles;
+});
+
+
+$mock->mock('cleanup', sub {
+    shift;
+    if (!$immutable) {
+        my $filename = shift;
+        $sourcefiles->delete($filename) || return undef;
+        return 1;
+    };
+    return undef;
+});
+
+
+$mock->mock('directory_exists', sub {
+    return $directory_exists;
+});
+
+
+=pod
+
+=head2 The source directory doesn't exist
+
+If the source directory doesn't exist, the error should be
+reported.
+
+=cut
+
+$immutable = 0;
```

Comment:
remove these lines and mocking above, just delete the directory using either

CAF::Path cleanup

```perl
$cmp->cleanup($SOURCE_DIR);
```
or Test::Quattor function
```perl
remove_any($SOURCE_DIR);
```

and that's it. remember, `CAF::Path` is mocked entirely in `Test::Quattor`. the only thing you need to do is to make the fake file/directory structure.